projects/plugins/protect/src/js/components/pricing-table/index.jsx

```diff
@@ -54,17 +54,26 @@ const ConnectedPricingTable = ( { onScanAdd } ) => {
 		onScanAdd();
 	} );
 
-	const getProtectFree = useCallback( () => {
+	const getProtectFree = useCallback( async () => {
 		recordEvent( 'jetpack_protect_connected_product_activated' );
 		setGetProtectFreeButtonIsLoading( true );
-		handleRegisterSite()
-			.then( () => setGetProtectFreeButtonIsLoading( false ) )
-			.then( () => {
-				refreshPlan();
-				refreshWaf();
-				refreshStatus( true );
-			} );
-	}, [ handleRegisterSite, recordEvent, refreshWaf, refreshPlan, refreshStatus ] );
+		try {
+			await handleRegisterSite();
+			startScanOptimistically();
+			await refreshPlan();
+			await refreshWaf();
+			await refreshStatus( true );
+		} finally {
+			setGetProtectFreeButtonIsLoading( false );
+		}
```

Comment:
After a site is registered, start scanning optimistically to ensure any existing scan status data (or error) is rendered while we refetch the status in the background.